lib/setup-opentofu.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// Node.js core
+const fs = require('fs').promises;
+const os = require('os');
+const path = require('path');
+
+// External
+const core = require('@actions/core');
+const tc = require('@actions/tool-cache');
+const io = require('@actions/io');
+const releases = require('@hashicorp/js-releases');
```

Comment:
@imjasonh Hey Jason! Unfortunately, the code you ported from the terraform repo does replicate setup-terraform action only: it downloads `terraform` binary from the **terraform registry** instead of fetching `tofu` binary from github. 😉 It appears that the only implemented change is renaming: the terraform binary is called opentofu. BTW it was agreed that the project is called OpenTofu whilst the binary is called `tofu` - please find details [here](https://github.com/opentofu/opentofu/issues/451).